packages/wrangler/src/deployment-bundle/module-collection.ts

```diff
@@ -284,46 +263,13 @@ export function createModuleCollector(props: {
 								// Check if this file is possibly from an npm package
 								// and if so, validate the import against the package.json exports
 								// and resolve the file path to the correct file.
-								if (args.path.includes("/") && !args.path.startsWith(".")) {
-									// get npm package name from string, taking into account scoped packages
-									const packageName = extractPackageName(args.path);
-									if (!packageName) {
-										throw new Error(
-											`Unable to extract npm package name from ${args.path}`
-										);
-									}
-									const packageJsonPath = path.join(
-										process.cwd(),
-										"node_modules",
-										packageName,
-										"package.json"
-									);
-									// Try and read the npm package's package.json
-									// and then resolve the import against the package's exports
-									// and then finally override filePath if we find a match.
-									try {
-										const packageJson = JSON.parse(
-											await readFile(packageJsonPath, "utf8")
-										);
-										const testResolved = resolveExports(
-											packageJson,
-											args.path.replace(`${packageName}/`, ""),
-											{
-												conditions: getBuildConditions(),
-											}
-										);
-										if (testResolved) {
-											filePath = path.join(
-												process.cwd(),
-												"node_modules",
-												packageName,
-												testResolved[0]
-											);
-										}
-									} catch (e) {
-										// We tried, now it'll just fall-through to the previous behaviour
-										// and ENOENT if the absolute file path doesn't exist.
-									}
+								try {
+									filePath = require.resolve(args.path, {
+										paths: [args.resolveDir],
+									});
```

Comment:
I think this might be better using esbuild's resolution algorithm rather than `require.resolve()`? Something along the lines of:

```ts
const resolved = await build.resolve(args.path, {
  kind: "import-statement",
  resolveDir: args.resolveDir,
  pluginData: {
	  skip: true,
  },
  });
  if (resolved.path) {
  filePath = resolved.path;
  }
```

with a check for `pluginData.skip` at the top of the `onResolve()` callback to prevent infinite loops.

Reply:
Updated to use esbuild's plugin. All the tests seem to pass. Is there a particular reason why you prefer the esbuild plugin vs require.resolve?